### DISCUSSION TITLE
Async form inputs is not editable (3.9.3+)

### DISCUSSION THREAD

**Discussion by arthurshlain** [alpinejs/alpine#2874] [7. Bugs (archived)]
In AlpineJS versions `3.9.3` - `3.10.2` async loaded form inputs with `x-model` is not editable for some reason. Any type of form inputs freezes in initial state - cannot edit text, choose select option etc. Focus was lost after any input change attempt.

In version `3.9.2` this issue not exist.

Minimal reproduction in link below.
https://codepen.io/ArtZ91/pen/ExozrVW

**Comment by SimoTod** [comment]
Prior v3.9.3 some code injected by x-html was never evaluated by alpine because x-html uses Alpine.mutateDom that pauses the mutation observer.
x-model should work as expected when dynamically injected (see https://codepen.io/SimoTod/pen/mdpZJoY).
In your case, it seems like the checkFormState function reverts all edits every time something changes (if you remove xx-init, everything works correctly).

**Reply by arthurshlain** [reply]
I don't quite understand what the cases when x-html is not evaluated. What are the limitations of mutateDom? Removing checkFormState function has no positive effect on real project, because there a lot of additional logic.

My code loads a set of rows asynch. Every row has a trigger (collapse open) to async load additional data with form. And inside this form can be additional async requests like typeahead, conditional select options etc.

**Reply by SimoTod** [reply]
It was something to do with x-html and x-for used together, i don't quite remember the details but that's what has been changed in that release. I'm not suggesting that you remove the function, I believe the logic there is somehow bugged and it's what breaks your example.
However, I haven't debugged your code since I haven't had time, so I might be wrong.

**Reply by arthurshlain** [reply]
Hi. I found my mistake. 

I write `this.formData = this.modelInfo.form;` 
instead of `this.formData = Object.assign({}, this.modelInfo.form);`
in my init() method.

Strange that in the 3.9.2 it worked successfully...
